src/notifications/notificationmanager.cpp

```diff
@@ -212,7 +345,52 @@ uint NotificationManager::Notify(const QString &appName, uint replacesId, const
                                  const QString &summary, const QString &body, const QStringList &actions,
                                  const QVariantHash &hints, int expireTimeout)
 {
-    NOTIFICATIONS_DEBUG("NOTIFY:" << appName << replacesId << appIcon << summary << body << actions << hints << expireTimeout);
+    uint id = 0;
+    if (!calledFromDBus()) {
+        id = handleNotify(getpid(), appName, replacesId, appIcon, summary, body, actions, hints, expireTimeout);
+    } else {
+        setDelayedReply(true);
+        ClientIdentifier *identifier = new ClientIdentifier(this, connection(), message());
+        connect(identifier, &ClientIdentifier::finished, this, &NotificationManager::identifiedNotify, Qt::QueuedConnection);
+    }
+    return id;
+}
+
+void NotificationManager::identifiedNotify(ClientIdentifier *identifier)
+{
+    QVariantList arguments(identifier->message().arguments());
+    QString appName = arguments.at(0).toString();
+    uint replacesId = arguments.at(1).toUInt();;
+    QString appIcon = arguments.at(2).toString();
+    QString summary = arguments.at(3).toString();
+    QString body = arguments.at(4).toString();;
```

Comment:
Nit: extra ; here and couple lines above.

Reply:
fixed